### ISSUE TITLE
Consider adding a `variable` and `apendVariables` option

### ISSUE THREAD

**Issue by pascalduez** [pascalduez/postcss-apply#15]
https://github.com/postcss/postcss-custom-properties#variables 
https://github.com/postcss/postcss-custom-properties#appendvariables  

In `postcss-apply` case that would maybe be named `sets`?

**Comment by kevinSuttle** [comment]
@pascalduez FWIW, the only way I know this would work is if you process and then hoist the output to be at the topmost style tag. 

<img width="1440" alt="screen shot 2016-09-28 at 4 14 12 pm" src="https://cloud.githubusercontent.com/assets/95672/18930612/ab7f2eac-8596-11e6-9b6c-3aac080d13aa.png">

**Comment by pascalduez** [comment]
@kevinSuttle Thanks for the feedback.
It might be that I need a good night of sleep, but I don't understand a damn thing at what you're trying to explain... Sorry.

**Comment by kevinSuttle** [comment]
Haha. No worries. If you'd like to do a screenshare or something when you're rested, we can. I'd like to invite @MoOx as well.

**Comment by mrtnbroder** [comment]
I really really really really need this!

I'm using webpack v2 + css modules + nextcss  for writing my css. Currently however, I have to import my `global.css` file (which holds all the `@apply` rule-sets) in every component-`styles.css` where I need these rules. This is very explict, but also very annoying. I'd like to have the `variables` or `sets` option here to have them available globally.

Any idea when you could work on this?

**Comment by pascalduez** [comment]
@mrtnbroder It's on my "priority list", I might need that feature soon as well ;)

**Comment by pascalduez** [comment]
Available as of `postcss-apply@0.6.0`. See #22 

Just wanted to push a first iteration so that you guys can play a bit with it and gives feedback hopefully. @mrtnbroder @AdamJo

I guess there will be things to improve/decide, like how it must work with the `preserve` option.

**Comment by pascalduez** [comment]
I don't want this to turn into a full blown inJS CSS stuff, but:

currently:
```js
const themes = {
  /* Set names won't be transformed, just `--` will be prepended. */
  'toolbar-theme': {
    /* Declaration properties can either be camel or kebab case. */
    backgroundColor: 'rebeccapurple',
    color: 'white',
    border: '1px solid green',
  },
};

[...]
postcss().use(apply({ sets: themes }))
[...]
```

```css
/* input */

.Toolbar {
  @apply --toolbar-theme;
}
```

Maybe we could go with the template trend:

```js
const themes = [
  `--toolbar-theme: { 
    background-color: rebeccapurple;
    color: white;
    border: 1px solid green;
  }`,
];

[...]
postcss().use(apply({ sets: themes }))
[...]
```
